src/TRestTrackViewerProcess.cxx

```diff
@@ -0,0 +1,189 @@
+/// Guide</a>.
+///
+///
+///--------------------------------------------------------------------------
+///
+/// RESTsoft - Software for Rare Event Searches with TPCs
+///
+/// \class      TRestTrackViewerProcess
+/// \author     Juanan Garcia
+///
+/// <hr>
+///
+#include "TRestTrackViewerProcess.h"
+
+ClassImp(TRestTrackViewerProcess);
+
+///////////////////////////////////////////////
+/// \brief Default constructor
+///
+TRestTrackViewerProcess::TRestTrackViewerProcess() { Initialize(); }
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestTrackViewerProcess::~TRestTrackViewerProcess() {}
+
+///////////////////////////////////////////////
+/// \brief Function to initialize input/output event members and define the
+/// section name
+///
+void TRestTrackViewerProcess::Initialize() {
+    SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
+    fTrackEvent = nullptr;
+    fSingleThreadOnly = true;
+}
+
+///////////////////////////////////////////////
+/// \brief Process initialization. It creates the canvas available in
+/// TRestEventProcess
+///
+void TRestTrackViewerProcess::InitProcess() {
+    CreateCanvas();
+    if (!hCanvas && fDrawHits)
+        hCanvas = new TCanvas("hitsCanvas", "hitsCanvas", fCanvasSize.X(), fCanvasSize.Y());
+}
+
+///////////////////////////////////////////////
+/// \brief The main processing event function
+///
+TRestEvent* TRestTrackViewerProcess::ProcessEvent(TRestEvent* evInput) {
+    // no need for verbose copy now
+    fTrackEvent = (TRestTrackEvent*)evInput;
+
+    fCanvas->cd();
+    TPad* pad = fTrackEvent->DrawEvent();
+    if (fDrawOriginEnd) {
+        GetOriginEnd(fAnalysisTree, originGr, endGr, legOrEnd);
+        DrawOriginEnd(pad, originGr, endGr, legOrEnd);
+    }
+
+    fCanvas->cd();
+    pad->Draw();
+    fCanvas->Update();
+
+    if (!fDrawHits) return fTrackEvent;
+
+    hCanvas->cd();
+    TPad* hPad = fTrackEvent->DrawHits();
+    if (fDrawOriginEnd) {
+        DrawOriginEnd(hPad, originGr, endGr, legOrEnd);
+    }
+
+    hCanvas->cd();
+    hPad->Draw();
+    hCanvas->Update();
+
+    return fTrackEvent;
+}
+
+///////////////////////////////////////////////
+/// \brief Get origin and end of the track if trackLineAnalysis have
+/// been performed
+///
+void TRestTrackViewerProcess::GetOriginEnd(TRestAnalysisTree* anaTree, std::vector<TGraph*>& origin,
+                                           std::vector<TGraph*>& end, std::vector<TLegend*>& leg) {
+    for (auto gr : origin)
+        if (gr) delete gr;
+
+    for (auto gr : end)
+        if (gr) delete gr;
+
+    for (auto l : leg)
+        if (l) delete l;
+
+    std::vector<double> originV(3, 0);
+    std::vector<std::string> oName = {"originX", "originY", "originZ"};
+    std::vector<double> endV(3, 0);
+    std::vector<std::string> eName = {"endX", "endY", "endZ"};
+
+    for (int i = 0; i < 3; i++) {
+        std::string fName = "alphaTrackAna_" + oName[i];
```

Comment:
`alphaTrackAna_` is defined by the user at the RML, then the name could be anything else, isnt?

Reply:
If this observable was extracted from `TRestTrackAnalysisProcess` one could perhaps retrieve the process name as follows:

```
fRunInfo->GetMetadataClass("TRestTrackAnalysisProcess")->GetName();
```

However, it might happen that there are two classes `TRestTrackAnalysisProcess` inside the data processing chain.

Not sure if there is a better way @nkx?

Reply:
I can see that `GetObservableValue` is used somewhere else in a similar way.

![image](https://user-images.githubusercontent.com/80903717/163877981-68411cfb-1f19-4828-9433-b9937e95e183.png)

I can think in some analysis processes that may use some variables from the analysis tree as input. So, I don't know what is the best solution here. As far as I understand the observables are always retreived from the `rml`. 

Perhaps we should create an issue about this?

Reply:
Yes, I dont think it is good to hardcode the process name,  since the user might change that value. On top of that, as I said earlier, it might happen that exist two `TRestRawSignalAnalysisProcess` inside the data processing chain. Which of them `TRestDetectorSingleChannelAnalysisProcess` would use?

On the other hand, the process `TRestDetectorSingleChannelAnalysisProcess` uses a `TRestDetectorSignalEvent`, so I believe the `max_amplitude`, `thr_integral` and `PeakAmplitudeIntegral` maps could be calculated directly inside the process without the need to retrieve them from a previous analysis. Isn't?

Reply:
Just created issue rest-for-physics/detectorlib#47